File: trivor-sdk/src/main/java/com/kalnee/trivor/sdk/models/Insight.java
package com.kalnee.trivor.sdk.models;

public class Insight<T> {

    private String code;
    private T value;

    public Insight() {
    }

    public Insight(String code, T value) {
        this.code = code;
        this.value = value;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public T getValue() {
        return value;
    }

    public void setValue(T value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return "Insight{" +
                "code='" + code + '\'' +
                ", value=" + value +
                '}';
    }
}

File: trivor-sdk/src/main/kotlin/com/kalnee/trivor/sdk/models/Insight.kt
package com.kalnee.trivor.sdk.models

class Insight<T> {

    var code: String? = null
    var value: T? = null

    constructor() {}

    constructor(code: String, value: T) {
        this.code = code
        this.value = value
    }

    override fun toString(): String {
        return "Insight{" +
                "code='" + code + '\'' +
                ", value=" + value +
                '}'
    }
}
